HISTORY.md

```diff
@@ -17,6 +17,7 @@
 * A batched MultiGet API (DB::MultiGet()) that supports retrieving keys from multiple column families.
 * Full and partitioned filters in the block-based table use an improved Bloom filter implementation, enabled with format_version 5 (or above) because previous releases cannot read this filter. This replacement is faster and more accurate, especially for high bits per key or millions of keys in a single (full) filter. For example, the new Bloom filter has a lower false positive rate at 16 bits per key than the old one at 100 bits per key.
 * Added AVX2 instructions to USE_SSE builds to accelerate the new Bloom filter and XXH3-based hash function on compatible x86_64 platforms (Haswell and later, ~2014).
+* Universal compaction now supports ttl-compactions when options.ttl is set. This would be the same as periodic compactions -- a full compaction will be triggered if any file is over the ttl threshold.
```

Comment:
I rethink about it, I think the current PR as it is, if we want to explain it, can be explained as that periodic compaction is that the whole DB is periodically full compacted.  The explanation that "a full compaction will be triggered if any file is over the ttl threshold" does not accurately explain the behavior, because, the TTL threshold uses the "creation time" timestamp, while we are using "file creation time" in current mode, and most likely it's the lowest level triggers compaction, not "any file".

Reply:
> The explanation that "a full compaction will be triggered if any file is over the ttl threshold" does not accurately explain the behavior

sure, makes sense, I can modify this.  

> the TTL threshold uses the "creation time" timestamp, while we are using "file creation time" in current mode, and most likely it's the lowest level triggers compaction, not "any file".

I didn't understand this part of your concern.
In the current PR we are no way depending on the "creation time" timestamp as we are falling back to the periodic compaction's implementation based on "file creation time".

Reply:
Yes that's my point. The PR as it is depends on "file creation time" so that we don't check a file is over the ttl threshold. Or do you mean the the creation time of the file is over the ttl threshold?

Reply:
Oh I see what you mean. I didn't want to put so much details in the history.md and wanted to keep it very high level, and hence I just said ttl. In this current case we would be checking if a file's "file creation time" is beyond the ttl threshold (as we reset periodic_comp_seconds to ttl). 

I think I'll just rephrase it like below to make it more clear. What do you think?

"Universal compaction now supports ttl-compactions when options.ttl is set. This would be the same as periodic compaction's implementation -- the ttl will be compared against the file_creation_time of the files. The whole DB will be fully compacted periodically."

Reply:
Or just say for universal compaction, makes ttl mean the same as periodic compaction.